src/main/java/org/jboss/logmanager/LogContext.java

```diff
@@ -529,18 +496,14 @@ public void close() throws Exception {
             for (AutoCloseable handler : closeHandlers) {
                 handler.close();
             }
-            synchronized (this) {
-                attachmentKey1 = null;
-                attachmentKey2 = null;
-                final var value1 = attachmentValue1;
-                attachmentValue1 = null;
-                if (value1 instanceof AutoCloseable) {
-                    ((AutoCloseable) value1).close();
-                }
-                final var value2 = attachmentValue2;
-                attachmentValue2 = null;
-                if (value2 instanceof AutoCloseable) {
-                    ((AutoCloseable) value2).close();
+            @SuppressWarnings("unchecked")
+            final Map<Object, Object> oldAttachments = (Map<Object, Object>) attachmentHandle.getAndSet(this, Map.of());
```

Comment:
If you use `Map<?, ?>` then you won't need the unsafe cast.

Reply:
Oh, duh. I'll fix that too.